src/qml/filters/gpstext/ui.qml

```diff
@@ -0,0 +1,815 @@
+				id: combo_smoothing
+				model: 
+					[   qsTr('0 (raw data)'), 					//0
+				   		qsTr('1 (interpolate missing data)'),	//1
+		                qsTr('3 points'),						//2
+		                qsTr('5 points (recommended)'),			//3
+		                qsTr('7 points'),						//4
+		                qsTr('11 points'),						//5
+		                qsTr('15 points'),						//6
+		                qsTr('31 points'),						//7
+		                qsTr('63 points'),						//8
+		                qsTr('127 points')						//9
+	                ]
+				Shotcut.HoverTip { text: qsTr('Smoothing is done by taking the average of X points.\nInterpolation is linearly done for missing values of altitude or heart rate.\nGPS data (speed, distance etc) computing is done only for smoothing > 0') }
+				currentIndex: 3
+				onActivated: {
+					switch (currentIndex) {
+						case 0:
+	                		onClicked: filter.set('smoothing_value', 0)
+							break
+						case 1:
+	                		onClicked: filter.set('smoothing_value', 1)
+							break
+						case 2:
+	                		onClicked: filter.set('smoothing_value', 3)
+							break
+						case 3:
+	                		onClicked: filter.set('smoothing_value', 5)
+							break
+						case 4:
+	                		onClicked: filter.set('smoothing_value', 7)
+							break
+						case 5:
+							onClicked: filter.set('smoothing_value', 11)
+							break;
+						case 6:
+							onClicked: filter.set('smoothing_value', 15)
+							break;
+						case 7:
+							onClicked: filter.set('smoothing_value', 31)
+							break;
+						case 8:
+							onClicked: filter.set('smoothing_value', 63)
+							break;
+						case 9:
+							onClicked: filter.set('smoothing_value', 127)
+							break;
+						default:
+							console.log('combo_smoothing: current index not supported: ' + currentIndex)
+					}
+				}
+			}
+			Shotcut.UndoButton {
+				onClicked: {
+					combo_smoothing.currentIndex = 3
+					filter.set('smoothing_value', 5)
+				}
+			}
+		}
+		
+		RowLayout {
+			Layout.columnSpan: 2
+			width: 250
+			
+			Label {
+				text: qsTr('Start processing at')
+				leftPadding: 10
+	            Layout.alignment: Qt.AlignRight
+				Shotcut.HoverTip { text: qsTr('GPS distances are calculated since the start of gps file, if you want to ignore the begining (for example when tracking a single lap) you can set here the time to start processing (UTC).') }
+	        }
+	    	TextField {
+	    		id: gps_processing_start
+	    		text: 'yyyy-MM-dd hh:mm:ss'
+	            horizontalAlignment: TextInput.AlignRight
+	            //TODO: regex to validate date yyyy-MM-dd hh:mm:ss
+	    		implicitWidth: 128
+				Shotcut.HoverTip { text: qsTr('Insert a date and time formatted as: YYYY-MM-DD HH:MM:SS (all fields mandatory), UTC timezone - same as GPS (use #gps_datetime_now# in filter to get current time).') }
+	        	onEditingFinished: filter.set('gps_processing_start_time', gps_processing_start.text);
+	    	}
+			Shotcut.UndoButton {
```

Comment:
There are multiple indentation alignment issues in this file that would be good to fix. Did you mix tabs and spaces?

Reply:
I used notepad++ before and switched to QT creator which seems to dynamically replace tabs with spaces. I changed it to tabs now and fixed most of the lines (there's still one remaining for next commit).

Reply:
Shotcut source uses spaces while MLT is tabs.

Reply:
Oh, so QT creator was right, I'll switch to spaces in the next commit.